src/wp-includes/class-wp-image-editor-imagick.php

```diff
@@ -320,6 +338,76 @@ public static function set_imagick_time_limit() {
 		}
 	}
 
+	/**
+	 * Gets the bit depth for PNG images and checks for indexed-color mode.
+	 *
+	 * Access the file directly, as we cannot currently rely on Imagick to identify
+	 * palette images with alpha support.
+	 *
+	 * @since 6.6.0
+	 */
+	protected function get_png_color_depth() {
```

Comment:
Imagick doesn't already provide a method for this?

Reply:
Ok, I am fine adding our own helper method. It would be good to add some more details to the inline comment, and ideally include a test with an image that fails detection without the helper.

Reply:
Well, unfortunately, when I test it now, the only images that IM fails to detect properly are non-paletted. I think there were more of them before, but I know I upgraded my local version of PHP at some point and messed around with newer versions of IM. So I'm not certain if those failing images are worth bothering about, or if I should try to downgrade PHP to double-check things?

Reply:
Alright, I'll see if I can test a few different PHP versions locally on my full set of PNG images and see if any paletted images fail. I was thinking perhaps I could try to add a unit test with getImageType, but I have a LOT more sample images locally so it'll be more thorough. Just gotta find the time to mess around with it again!